packages/core/core-flows/src/payment/steps/capture-payment.ts

```diff
@@ -36,6 +36,26 @@ export const capturePaymentStep = createStep(
 
     const payment = await paymentModule.capturePayment(input)
 
-    return new StepResponse(payment)
+    return new StepResponse(payment, payment)
+  },
+  async (payment, { container }) => {
+    if (!payment) {
+      return
+    }
+
+    const paymentModule = container.resolve<IPaymentModuleService>(
+      Modules.PAYMENT
+    )
+
+    // Every time we capture a payment, we create a new capture. We want to delete the newset capture if we capture a payment again.
```

Comment:
In which circumstances we can capture a payment more than once?

Reply:
Based on the implementation in the payment module and the models, a payment can have multiple partial captures, so this just implements it with that logic in mind. For example, you might have 100 EUR authorized, but you only want to eg. capture 50 before shipping, and the other 50 after shipping.

Reply:
Yeah, that part I get.
But as we are deleting the capture, I'm not sure how that could be possible other than refunding it.

Reply:
Ah you refer more to Oli's comment then, yes I realized it is probably a mistake to delete the capture from the DB, I'll revert that.